.devcontainer/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim-bullseye AS base
+
+RUN set -x
+
+RUN apt-get update \
+    && apt-get -y install curl \
+    && apt-get -y install net-tools \
+    && apt-get -y install procps \
+    && apt-get -y install build-essential \
+    && apt-get -y install docker.io
+
+FROM base AS promptflow-package
+# specific pip version to workaround: https://github.com/conda/conda/issues/10178
+RUN pip install --upgrade pip==20.1.1
+RUN pip install promptflow[azure] --upgrade \
+    && pip install promptflow-tools --upgrade \
+    && pip install python-dotenv --upgrade \
+    && pip install bs4 --upgrade
```

Comment:
1. please refer to script/building folder and our building pipelines.
2. we should not direct pip install promptflow and tools, we need a separate build step/build command.

Target is not only run pf.
We should provide container to just build pf and test pf.
environment should be just set.

By the way conda image maybe a better base image than python.

Reply:
Extend usage: Dev container can also allow pre-installed github secrets

Reply:
About "pre-installed github secrets",
I think most developers download promptflow projects using HTTPS instead of SSH? So shouldn't secrets be used?